js/cex.js

```diff
@@ -1507,18 +1520,89 @@ module.exports = class cex extends Exchange {
         const request = {
             'currency': currency['id'],
         };
-        const response = await this.privatePostGetAddress (this.extend (request, params));
-        const address = this.safeString (response, 'data');
+        const [ networkId, query ] = this.handleNetworkCodeAndParams (code, params);
+        const response = await this.privatePostGetCryptoAddress (this.extend (request, query));
+        //
+        //    {
+        //         "e": "get_crypto_address",
+        //         "ok": "ok",
+        //         "data": {
+        //             "name": "BTC",
+        //             "addresses": [
+        //                 {
+        //                     "blockchain": "Bitcoin",
+        //                     "address": "2BvKwd7UwrdTjx2nzhscFYXwqCjCaaHJmW"
+        //                 }
+        //             ]
+        //         }
+        //     }
+        //
+        const data = this.safeValue (response, 'data', {});
+        const addresses = this.safeValue (data, 'addresses', []);
+        let chosenNetworkId = undefined;
+        const chainsIndexedById = this.indexBy (addresses, 'blockchain');
+        if (networkId !== undefined) {
+            if (networkId in chainsIndexedById) {
+                chosenNetworkId = networkId;
+            } else {
+                throw new InvalidAddress (this.id + ' fetchDepositAddress() - ' + code + ' no deposit networks were found for ' + networkId);
+            }
+        } else {
+            const ids = Object.keys (chainsIndexedById);
+            const defaultNetwordId = this.defaultNetworkId (code);
+            chosenNetworkId = (defaultNetwordId in chainsIndexedById) ? defaultNetwordId : ids[0];
+        }
+        const addressObject = chainsIndexedById[chosenNetworkId];
+        const networkCode = this.networkIdToCode (chosenNetworkId);
+        const address = this.safeString (addressObject, 'address');
         this.checkAddress (address);
         return {
             'currency': code,
             'address': address,
             'tag': undefined,
-            'network': undefined,
-            'info': response,
+            'network': networkCode,
+            'info': addressObject,
         };
     }
 
+    networkIdToCode (networkId) {
```

Comment:
@ttodua so we define the exact same methods in bybit too, if we're going to use this approach we should move them to Exchange.js no?

Reply:
> @ttodua so we define the exact same methods in bybit too, if we're going to use this approach we should move them to Exchange.js no?

yes, right, I was already going to submit them in base, but at first, made this PR to demonstrate that we will need to use those methods in most implementation.
I'll modify this PR and ask your re-review tomorrow.

Reply:
@carlosmiei , after thorough revision and planning, I have submitted the most optimal and unified methods that are reusable (can show other existing exchanges too which will benefit from these methods, even the above mentioned 'bybit').
If you look into them thoroughly, you understand what each of those methods do and all of them is important in the process of network-unification across lib.

Reply:
I've attached example responses in the top-post.

Reply:
I think this PR might be needed soon, because `networkCodeToId` and other methods will be also necessary for @AlgoQ / @samgermain  to work on network-related slugs (in fetchTransactionFees, which is underway).